### DISCUSSION TITLE
Setup never ends and reference image most times not shown

### DISCUSSION THREAD

**Discussion by labmaster** [jomjol/AI-on-the-edge-device#633] [Q&A]
Version: Rolling (2022-01-27) 
bootloader.bin, partitions.bin, firmware.bin and full SD-card redone
Checked that "config.ini  <--   ImageQuality = 12"  is set

Problem:
After going thru all the setup steps the system still keeps staying in the setup process.
Even after reboot it starts over the setup from the beginning, most times not showing the reference images then.
Needs a lot of reloads to sometimes gets the taken reference images shown correctly.

Any Hints ?

**Comment by jomjol** [comment]
Did you really finish the setup with the following button?
![grafik](https://user-images.githubusercontent.com/30766535/152291437-8132bbd2-449d-4511-9be2-68fc51747ce0.png)

**Reply by labmaster** [reply]
Yes, sure,  every time i went thru the whole setup process (cant count anymore how often i have tried)  this was my last step, clicking at this button.
But on the next page reload i ended in the first step of the setup

**Comment by friedpa** [comment]
@labmaster 
In the config.ini: is the switch `SetupMode` set to false?
![grafik](https://user-images.githubusercontent.com/42890695/152291807-e1799b4d-0a4f-4aee-ac10-cfdb6a64cd4f.png)

**Reply by labmaster** [reply]
It seems this might be the right direction to search for the problem.

-The SDCard itself seems to be accessible as my wlan.ini setting takes effect.
-When i unpower the device, remove the SdCard and read the content eg. of /config/config.ini on my PC , the config.ini seems to be completely cleared. even "ImageQuality" what i made sure to be set to 12 is now ";ImageQuality ="
All other entry's even now at ";xxxx...=", ONLY the last one "SetupMode = false" is now without a ; at the beginning and set to "false"

EDIT:
A minute ago i checked with the last try round and now the SDCard content seems to be OK , only the "SetupMode = true" keeps staying on true.
This kind of debugging is not so easy, as the physical device (where it is mounted on the watermeter is quite a distance away.

**Comment by labmaster** [comment]
After 50 or more reboots and UI setup tries, it now happens that the settings got stored correctly and the SetupMode" is true .
I finally now into the main operation screen.

But now the next challenge is waiting for me :-)

**Comment by friedpa** [comment]
You can set the `SetupMode` in the config.ini via `Expert Mode` -> `Edit Config.ini direct` -> last Section in the Config.ini file. Set it to `False `or mark it out with an Semicolon at the beginning of the row.
As far as I rember the Config.ini is writen to the SD-Card several times. If the SD-Card gives you already in the beginning such troubles just try another one. 4GByte with FAT32 is fine. SD-Cards are the most critical issue (after the reboot issue). I tried at least 4-5 different ones. Now my system is absolutely stable.

**Comment by labmaster** [comment]
I can not setup.ini via  Expert Mode -> Edit Config.ini direct as long as i am locked to the setup mode.
In the last 2 Days i already tried 4 different SDCards, 3 x 16GB and 1 x 32GB, i did not had any smaller and  trying to order 8GB or below gave me hard times at least within decent time, so i gave up.

**Comment by jomjol** [comment]
Sorry, there is no other way around, than a running SD-card. I have also a 32 GB card running. Unfortunately there is no rule, how to find out, if this works. 
One user was succesfull, by formating a 32 GB with only 4 GB.

**Comment by VinceW31** [comment]
Switch off your Virus checker.  Mine (Kaspersky) sometimes blocked some of the HTML from loading and that stopped any progress.

**Reply by nicmela** [reply]
I confirm
